### DISCUSSION TITLE
More fields for Lean-to

### DISCUSSION THREAD

**Discussion by ERYpTION** [Zverik/every_door#794] [Ideas]
Would it be possible to add a couple of fields to the 'More Fields' option when editing a Lean-To (a Shelter-type)? 

Where I live operators of Lean-tos often recommend doing a reservation and sometimes require paying a small fee for use. Therefore it would be useful with easy access to adding this information with the Fee and Reservation fields. They are currently missing for Lean-tos.

Thanks.

**Comment by mnalis** [comment]
I could be wrong, but I think you should request that at [id-tagging-schema](https://github.com/openstreetmap/id-tagging-schema) which [Every Door uses for tagging presets](https://github.com/Zverik/every_door/blob/main/README.md#presets-and-translations).


Particularly, a change to https://github.com/openstreetmap/id-tagging-schema/blob/main/data/presets/amenity/shelter/lean_to.json adding `moreFields` of `fee`, like e.g. done for [verterinary](https://github.com/openstreetmap/id-tagging-schema/blob/ce216355bd9a62bc20dfe56f0417ec7bcdb91647/data/presets/amenity/veterinary.json#L13).

In fact, as it is relatively simple change of JOSN (text) file, you may even want to try doing a PR yourself there? It's relatively easy (fork the repository, and click on edit button, and when you've done your change if it looks OK click on "create pull request" when it pops out). 

Or of course you can simply [open the issue there](https://github.com/openstreetmap/id-tagging-schema/issues) and wait for someone to pick it up. Feel free to mention that issue URL here, so interested parties can find it more easily (and perhaps help)

**Reply by ERYpTION** [reply]
I see an objection to another request for more fields because of low usage. This will probably also be an issue in this case then. Apparently it is not normal for Lean-Tos asking for reservation and payment in other places.

**Reply by mnalis** [reply]
Well yeah, I've never heard of such practice eitheer, not for `shelter_type=lean_to` (for basic shelter, maybe). You are correct, it seems extremely little used ([only 169](https://overpass-turbo.eu/s/1RQj) in the whole world have `fee=yes`, out of [17585](https://taginfo.openstreetmap.org/tags/shelter_type=lean_to#overview) lean-to's - that is less than `1%`, so unlikely to be adopted (by id-tagging-presets or elsewhere)...

However, when(/if) https://github.com/Zverik/every_door/issues/784 is implemented, you should be able to make such custom field for yourself, though! I'm excited about that one too!

In the meantime, you can always use **raw tag** editor feature of EveryDoor (click on that "table" icon in upper right when editing a POI) and add `fee=yes` manually to those lean-to's.

**Reply by ERYpTION** [reply]
Here in Denmark about 2/3 of the lean-tos on public land can actually be booked in advance and for the privately owned (but publicly available) lean-tos I think it is about the same. Many of the privately owned lean-tos also require a small fee. So even though a tag is rarely used (reservation=yes/required) the actual situation can be different in some areas.

**Reply by tordans** [reply]
FYI, we added some guidelines to the id-tagging-schema so have a shared understanding on what to add and how to get there https://github.com/openstreetmap/id-tagging-schema/blob/main/GUIDELINES.md

Usage numbers are one factor. Relevant usage in apps or something like this is another. – See the Link for more.

**Reply by mnalis** [reply]
So @tordans if I understand correctly, that _"🏝️ Local Presets and Fields"_ would imply it would be OK to modify `shelter_type=lean_to` to only have `fee=*` field in Denmark (and Sweden)? 

It says _"presets and fields can be given a local filter"_ so it seems to be possible; but searching for "filter" does not give any more matches... 
I'm aware _whole preset_ can be made per-country (via `locationSet` as in e.g. [moped_link-NL.json](https://github.com/openstreetmap/id-tagging-schema/blob/main/data/presets/highway/cycleway/moped_link-NL.json), but how can ***only some field*** in the preset be limited to a country, could you point me to example or docs? If so, I'll try making a PR for this.

**Reply by mnalis** [reply]
I seem to have found example at [presets/highway/crossing.json](https://github.com/openstreetmap/id-tagging-schema/blob/main/data/presets/highway/crossing.json) which references [@templates/crossing/markings.json](https://github.com/openstreetmap/id-tagging-schema/blob/main/data/presets/%40templates/crossing/markings.json) which references [fields/crossing/markings-DE-AT-CH.json](https://github.com/openstreetmap/id-tagging-schema/blob/main/data/fields/crossing/markings-DE-AT-CH.json) which seems to has `locationSet` which is hopefully applied for that field only. Seems somewhat complex, but I can try to play with it...

_**Update:**_ Apparently, `locationSet` can also be used directly on the field? https://github.com/ideditor/schema-builder#locationset-1. See also discussion here: https://github.com/ideditor/schema-builder/issues/94#issuecomment-1828503402 and [fields/royal_cypher-GB.json](https://github.com/openstreetmap/id-tagging-schema/blob/main/data/fields/royal_cypher-GB.json) + [presets/amenity/post_box.json](https://github.com/openstreetmap/id-tagging-schema/blob/7629cf2bd8b5b2bd0b1b9356990c7bc3c395cad9/data/presets/amenity/post_box.json#L8) as an another (simpler) example

**Reply by mnalis** [reply]
- ref. https://github.com/openstreetmap/id-tagging-schema/issues/1350
